CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+
+Only code in `master` should be deployed to production, while code in other branches may be deployed for testing purposes.
+
+### Scope
+
+A fix, feature or issue branch should be **small and focused** and should be scoped to a **single specific task**. Do not combine new features and refactoring of existing code.
+
+### Rebasing, Merging
+
+- `fix/`, `feature/`, `issue/` branches are merged into `develop`
+  - These branches may be rebased on `develop` in order to clean up the commit history
+- `develop` branch is merged into `master` later by the maintainers of the project
+  - `develop` must not ever be rebased
+
+[git rebase](https://git-scm.com/docs/git-rebase) is often used before merging a branch. If you are not familiar with rebasing, you should read [this](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) guide. It is also used in order to [clean up](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) your commit trail before merging to `develop`.
+
+## Commit messages guideline
+
+The first line of a commit message is the subject. The commit message may contain a body, separated from the subject by an empty line.
+
+### Subject
+
+The subject contains the concerning component or topic (if applicable) and a concise message in [the imperative mood](https://chris.beams.io/posts/git-commit/#imperative), starting with a capital. The subject may also contain references to issues or other resources.
+
+The component or topic is typically a few characters long and should always be present. Component names are:
+
+* `doc`: documentation
+* `sch`: changes to schematic (PIN mapping, HW connection changes,...)
+* `pcb`: changes to device PCB and routing
+* `hw` : changes to the design (adding new sensor, IC, etc.)
+* `dev`: other non-functional development changes, e.g. Makefile, .gitignore, editor config
+* `com `: changes to communication ports (UART, I2C, SPI,...)
+* `tmr `: changes to timers, counters components
+* `compiler`: compiler related changes (attributes, commands,..)
+* `lib`:  libraries (additions, changes, improvements, fixes)
+* `app`: application level changes (additions, improvements, fixes)
+* `all`: changes affecting all code
+
+Changes that affect multiple components can be comma separated.
+
+Good commit messages:
+
+* `doc: Add README with build instructions`
+* `dev: Improve CMAKE build with debug and release options`
+* `lib,com,app: Fix flash memory read API`
+
+Make sure that commits are scoped to something meaningful and could, potentially, be merged individually.
+
+### Body
+
+The body may contain a more detailed description of the commit, explaining what it changes and why. The "how" is less relevant, as this should be obvious from the diff.
+
+## Legal
+
+Any contributions related to this repository weather it might be source code improvements or design changes require signing the [Contributor License Agreement (CLA)](https://cla-assistant.io/TheThingsIndustries/generic-node-se).
```

Comment:
```suggestion
Any contributions related to this repository whether it might be source code improvements or design changes require signing the [Contributor License Agreement (CLA)](https://cla-assistant.io/TheThingsIndustries/generic-node-se).
```

Reply:
Done, sorry for that